aten/src/ATen/native/Distributions.cpp

```diff
@@ -231,7 +232,34 @@ struct ExponentialStub {
 };
 
 Tensor& exponential_(Tensor& self, double lambda, c10::optional<Generator> gen) {
-  return at::native::templates::exponential_impl_<ExponentialStub, Generator>(self, lambda, std::move(gen));
+  at::native::templates::exponential_impl_<ExponentialStub, Generator>(self, lambda, std::move(gen));
+  #if (AT_MKL_ENABLED() && !USE_CUDA) 
+  {
+    at::Tensor zero_idx;
+    int64_t n_zeros;
+    bool has_zero;
+
+    zero_idx = (self == 0.).nonzero();
```

Comment:
if https://github.com/pytorch/pytorch/pull/101720/commits/a700ccc5a37ab619ab29f5315f7f3f0f314ea4db can fix this issue, then it is fine.

the fix from this PR is going to be a lot slower since it scans the whole tensor multiple times, this PR can be further optimized if we plan the memory access more carefully, but it is still going to be slower since you have to scan whether the output tensor (if has zero) anyway.